neuro_data/pipeline_management/NeuroDataPipelineManagement.py

```diff
@@ -0,0 +1,172 @@
+                        
+                    self.manually_insert_area_for_scan(target_scan, area)
+                elif user_input == 'n':
+                    return
+        else:
+            print('[Preprocessing Check]: AreaMembership Check Passed')
+
+        # Check if neuron layers are labeled
+        if len(self.pipeline_anatomy.LayerMembership() & target_scan) == 0:
+            print('[Preprocessing Check]: ' + str(target_scan) + " LayerMembership is not populated")
+
+            user_input = None
+            while user_input not in ['y', 'n']:
+                user_input = input('Should we manually insert if the layer are known and all the same? [y/n/exit]')
+                
+                if user_input == 'y':
+                    layer = input('Input layer to label neurons with:')
+                    while layer not in ['L1', 'L2/3', 'L4']:
+                        if layer == 'exit':
+                            return
+                        print('Invalid Layer!')
+                        layer = input('Input layer to label neurons with:')
+                        
+                    
+                    self.manually_insert_layer_for_scan(target_scan, layer)
+                elif user_input == 'n':
+                    return
+        else:
+            print('[Preprocessing Check]: LayerMembership Check Passed')
+
+        # Check pipeline_stimulus.Sync() table
+        if len(self.pipeline_stimulus.Sync() & target_scan) == 0:
+            print('[Preprocessing Check]: ' + str(target_scan) + ' pipeline_stimulus.Sync() table is not processed or failed to processed')
+            return
+        else:
+            print('[Preprocessing Check]: ScanDone Check Passed')
+
+        # All tables requirements are met, begin neurodata dataset population
+        print('[Preprocessing Check]: All table requirements passed, beginning neuro_data populating:')
+        
+        # Get the ScanDone primary key reference
+        target_scan_done_key = (self.pipeline_fuse.ScanDone() & target_scan).fetch1('KEY')
+
+        # Insert into StaticScanCandidate
+        if len(StaticScanCandidate & target_scan_done_key) == 0:
+            StaticScanCandidate.insert1(target_scan_done_key)
+            print('[NeuroData.Static Populate]: Successfully inserted Scan into StaticScanCandidate')
+        else:
+            print('[NeuroData.Static Populate]: Scan has already been added to StaticScanCandidate')
+
+        # Populating StaticScans
+        print("[NeuroData.Static Populate]: Populating StaticScan:")
+        StaticScan().populate(target_scan_done_key)
+
+        # Populate ConditionTier
+        print("[NeuroData.Static Populate]: Populating ConditionTier:")
+        ConditionTier.populate(target_scan_done_key)
+
+        # Populate Frame
+        print("[NeuroData.Static Populate]: Populating Frame:")
+        Frame.populate(dict(preproc_id = 0))
+
+        # Populate InputResponse
```

Comment:
Historically, we check if there is any trials having wrong flip numbers, add those trials to ExcludedTrial and populate InputResponse afterward. I am not sure if the bug causing wrong flip numbers has been fixed or not. Maybe add those checks just to be safe.
Codes used before to check flip times:
```
unprocessed = (fuse.ScanDone & target_scan) - data.InputResponse
keys, flips = (stimulus.Trial & stimulus.Frame & unprocessed - data.ExcludedTrial).fetch('KEY', 'flip_times')
n_flips = np.array([f.shape[1] for f in flips])
# There should only be 3 frames for stimulus.Frame stimulus.
correct_flips = 3
bad_trials = (n_flips != correct_flips)
bad_trial_keys = list(compress(keys, bad_trials))
data.ExcludedTrial.insert(bad_trial_keys)
```

Reply:
Thanks, I add it in sometime later tomorrow

Reply:
@ZhuokunDing I added the code to the pull request.